fix: validate tx related to event before storing in PXE storage

mverzilli opened the pull request:
Aztec-nr delivers events to PXE together with a txHash. However we were not validating that the denoted tx exists, that it indeed emitted the event, and that the PXE is caught up to the block where the tx was included.

Fixes F-114

benesjan reviewed (approved):
Very nice and thorough. Thanks

nventuro commented on yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts:
```diff
@@ -795,9 +795,22 @@ export class PXEOracleInterface implements ExecutionDataProvider {
+
+    if (!txEffect || txEffect.l2BlockNumber > syncedBlockNumber) {
+      throw new Error(`Could not find tx effect for tx hash ${txHash}`);
+    }
```
I'd throw two different errors - the tx not being could could indicate the hash being bad or a reorg having occurred. The tx existing but with a larger block number is a different kind of problem. Would likely be useful for diagnosing if someone runs into this in the wild.

nventuro commented on yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts:
```diff
@@ -795,9 +795,22 @@ export class PXEOracleInterface implements ExecutionDataProvider {
+    const eventInTx = txEffect.data.nullifiers.some(nullifier => nullifier.equals(siloedEventCommitment));
+    if (!eventInTx) {
+      throw new Error(
+        `Event commitment ${eventCommitment} (siloed as ${siloedEventCommitment} is not present in tx ${txHash})`,
```
```suggestion
        `Event commitment ${eventCommitment} (siloed as ${siloedEventCommitment}) is not present in tx ${txHash}`,
```

nventuro reviewed (approved):
Lovely tests, very nice.

mverzilli committed:
fix: validate tx related to event before storing in PXE storage

Aztec-nr delivers events to PXE together with a txHash. However we were not validating that the denoted tx exists, that it indeed emitted the event, and that the PXE is caught up to the block where the tx was included.

Fixes F-114
